src/color.rs

```diff
@@ -262,6 +258,48 @@ pub trait ColorComponentParser<'i> {
             ref t => return Err(location.new_unexpected_token_error(t.clone())),
         })
     }
+
+    /// Parse a `<hue>` value.
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#typedef-hue
+    fn parse_hue<'t>(
+        &self,
+        input: &mut Parser<'i, 't>,
+    ) -> Result<Hue, ParseError<'i, Self::Error>> {
+        let location = input.current_source_location();
+        Ok(match *input.next()? {
+            Token::Number { value: degrees, .. } => Hue { degrees },
```

Comment:
So the reason this (`parse_angle_or_number`, etc) are in `ColorComponentParser` is so that we can add `calc()` support externally e.g., in Firefox. However `Hue` is not constructible outside of this crate so that's not feasible. That needs to at least be fixed